server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1217,6 +1045,62 @@ public JobDetail refreshPools(
     }
 
     /**
+     * Creates a custom pool for an Owner. Floating pools are not tied to any
+     * upstream subscription, and are most commonly used for custom content
+     * delivery in Satellite.
+     * Also helps in on-site deployment testing
+     *
+     * @return a Pool object
+     * @httpcode 404
+     * @httpcode 200
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{owner_key}/pools")
+    public Pool createPool(@PathParam("owner_key") @Verify(Owner.class) String ownerKey, Pool pool) {
+
+        log.info("Creating custom pool for owner {}: {}" + ownerKey, pool);
+
+        // Correct owner & products
+        Owner owner = findOwner(ownerKey);
+        pool.setOwner(owner);
+
+        pool = resolverUtil.resolvePool(pool);
+        return poolManager.createAndEnrichPools(pool);
+    }
+
+    /**
+     * Updates a pool for an Owner.
+     * assumes this is a normal pool, and errors out otherwise cause we cannot
+     * create master pools from bonus pools
+     * TODO: while this method replaces the now deprecated updateSubsciption, it
+     * still uses it underneath. We need to re-implement the wheel like we did
+     * in createPool
+     *
+     * @httpcode 404
+     * @httpcode 200
+     */
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{owner_key}/pools")
+    public void updatePool(@PathParam("owner_key") @Verify(Owner.class) String ownerKey,
+            Pool pool) {
+
+        if (pool.getType() != PoolType.NORMAL) {
+            throw new BadRequestException(i18n.tr("Cannot update bonus pools, as they are auto generated"));
+        }
+
+        // convert to subscription so we can use updateSubscription
+        Subscription subscription = new Subscription(pool.getOwner(), pool.getProduct(),
```

Comment:
@Ceiu I agree, hence the TODO on the API. To do so will mean a few more ripple effect changes ( like the changes incurred as a part of changing createPools ) , and given the fact this is a test only API used in far fewer places than createPools ( only when tests need to modify upstream subscription source ), I thought we should tackle this later.